File: game-manager-service/src/main/java/com/stackroute/quizify/gamemanager/domain/Game.java
package com.stackroute.quizify.gamemanager.domain;


import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;



@Document(collection="game")
@Data
public class Game {
    @Id
    private long id;
    private String name;
    private Category category;
    private Topic topic;
    private Genre genre;
    private Tag tag;
    private String level;
    private String imageUrl;
    private int numOfQuestion;
    private List<Question> questions;
    private int timeDuration;
    private int liked;
    private int playCount;
    private List<String> rules;
}

File: Quizify-FrontEnd/src/app/quizify/tsclasses/game.ts
import { Category } from './category';
import { Question } from './question';
import { Admin } from './admin';
import { Topic } from './topic';
import { Genre } from './genre';

export class Game {
     id: number;
     name: string;
     category: Category;
     topic: Topic;
     playcount: number;
     numOfQuestion: number;
     questions: Array<Question>;
     admin: Admin;
     genre: Genre;
     level: string;
     liked: number;
     timeDuration: number;
}

